REGRESSION(265930@main): [ iOS 16 ] TestWTF.WTF_Packed.PackedAlignedPtr is a constant failure

justinmichaud opened the pull request:
#### 00c679a9ae1b56c9a4e21f43aaab9bb26123222c
<pre>
REGRESSION(265930@main): [ iOS 16 ] TestWTF.WTF_Packed.PackedAlignedPtr is a constant failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=259366">https://bugs.webkit.org/show_bug.cgi?id=259366</a>
rdar://112620141

Reviewed by Brent Fulgham.

I missed a compile guard.

* Tools/TestWebKitAPI/Tests/WTF/Packed.cpp:
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/266263@main">https://commits.webkit.org/266263@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2fbdd1429d1af7b5cbff673ed1f23147526d863c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13361 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/13675 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14007 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15098 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/12721 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/13440 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16182 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/13701 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15402 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13528 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14184 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11301 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/15738 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11472 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/12056 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/19110 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12547 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12223 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/15440 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12728 "Built successfully") | [💥 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10585 "An unexpected error occured. Recent messages:OS: Ventura (13.4.1), Xcode: 14.3; Skipping applying patch since patch_id isn't provided; Checked out pull request") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/11998 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16321 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1527 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12569 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Tools/TestWebKitAPI/Tests/WTF/Packed.cpp:
```diff
@@ -96,7 +96,7 @@ TEST(WTF_Packed, PackedAlignedPtr)
     {
         PackedAlignedPtr<uint8_t, 16> key { nullptr };
-#if (OS(IOS) || OS(TVOS) || OS(WATCHOS)) && CPU(ARM64)
+#if (OS(IOS) || OS(TVOS) || OS(WATCHOS)) && CPU(ARM64) && !PLATFORM(IOS_FAMILY_SIMULATOR)
```
This might need OS(VISION), as well. I assume you are not using IOS_FAMILY because this isn't the right size for MACCATALYST.

brentfulgham reviewed (approved):
Looks good, but please consider whether OS(VISION) is needed, too.

Commit:
REGRESSION(265930@main): [ iOS 16 ] TestWTF.WTF_Packed.PackedAlignedPtr is a constant failure
https://bugs.webkit.org/show_bug.cgi?id=259366
rdar://112620141

Reviewed by Brent Fulgham.

I missed a compile guard.

* Tools/TestWebKitAPI/Tests/WTF/Packed.cpp:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/266263@main
